backends/graphics/atari/atari-graphics.h

```diff
@@ -0,0 +1,272 @@
+	void setMouseCursor(const void *buf, uint w, uint h, int hotspotX, int hotspotY, uint32 keycolor, bool dontScale = false, const Graphics::PixelFormat *format = NULL) override;
+	void setCursorPalette(const byte *colors, uint start, uint num) override;
+
+	Common::Point getMousePosition() const { return _cursor.getPosition(); }
+	void updateMousePosition(int deltaX, int deltaY);
+
+	bool notifyEvent(const Common::Event &event) override;
+	Common::Keymap *getKeymap() const;
+
+protected:
+	const Graphics::PixelFormat PIXELFORMAT8 = Graphics::PixelFormat::createFormatCLUT8();
+	const Graphics::PixelFormat PIXELFORMAT16 = getOverlayFormat();
+
+	bool allocateAtariSurface(byte *&buf, Graphics::Surface &surface,
+							  int width, int height, const Graphics::PixelFormat &format,
+							  int mode, uintptr mask = 0x00000000);
+
+	bool _vgaMonitor = true;
+
+	enum class GraphicsMode : int {
+		DirectRendering,
+		SingleBuffering,
+		DoubleBuffering,
+		TripleBuffering
+	};
+
+	struct GraphicsState {
+		bool operator==(const GraphicsState &other) const {
+			return mode == other.mode
+				&& width == other.width
+				&& height == other.height
+				&& format == other.format;
+		}
+		bool operator!=(const GraphicsState &other) const {
+			return !(*this == other);
+		}
+
+		GraphicsMode mode;
+		int width;
+		int height;
+		Graphics::PixelFormat format;
+	};
+	GraphicsState _pendingState = {};
+
+	static const int SCREENS = 3;
+	static const int FRONT_BUFFER = 0;
+	static const int BACK_BUFFER1 = 1;
+	static const int BACK_BUFFER2 = 2;
+	byte *_screen[SCREENS] = {};	// for Mfree() purposes only
+	byte *_screenAligned[SCREENS] = {};
+	Graphics::Surface _screenSurface;
+
+	byte *_chunkyBuffer = nullptr;	// for Mfree() purposes only
+	Graphics::Surface _chunkySurface;
+
+	byte *_overlayScreen = nullptr;	// for Mfree() purposes only
+	Graphics::Surface _screenOverlaySurface;
+
+	byte *_overlayBuffer = nullptr;	// for Mfree() purposes only
+	Graphics::Surface _overlaySurface;
+
+private:
+	enum CustomEventAction {
+		kActionToggleAspectRatioCorrection = 100,
+	};
+
+	void setVidelResolution() const;
+	void waitForVbl() const;
+
+	bool updateOverlay();
+	bool updateDirectBuffer();
+	bool updateSingleBuffer();
+	bool updateDoubleAndTripleBuffer();
+
+	virtual void* allocFast(size_t bytes) const = 0;
+
+	virtual void copySurfaceToSurface(const Graphics::Surface &srcSurface, Graphics::Surface &dstSurface) const = 0;
+	virtual void copyRectToSurface(const Graphics::Surface &srcSurface, int destX, int destY,
+								   Graphics::Surface &dstSurface, const Common::Rect &subRect) const = 0;
+	virtual void copyRectToSurfaceWithKey(const Graphics::Surface &srcSurface, int destX, int destY,
+										  Graphics::Surface &dstSurface, const Common::Rect &subRect, uint32 key) const = 0;
+	virtual void alignRect(const Graphics::Surface &srcSurface, Common::Rect &rect) const {}
+	virtual void sync() const {}
+
+	enum class ScaleMode {
+		None,
```

Comment:
This doesn't conform to our code naming conventions. But since it is super-local, I do not insist.